plugins/wdc/wdc-nvme.c

```diff
@@ -1038,6 +1039,54 @@ static int wdc_create_log_file(char *file, __u8 *drive_log_data,
 	return 0;
 }
 
+static const uint8_t zero_uuid[16] = { 0 };
+
+static int get_wdc_uuid_index(int fd, __u8* uuid_ix, uuid_t uuid)
+{
+	*uuid_ix = 1;
+	int ret = -1;
+
+	struct nvme_id_ctrl ctrl;
+	struct nvme_id_uuid_list uuid_list;
+
+	/* Determine if the controller supports a UUID List by checking CTRATT */
+	ret = nvme_identify_ctrl(fd, &ctrl);
+
+	if (ret == 0) {
+
+		if ((ctrl.ctrattr & NVME_CTRL_CTRATT_UUID_LIST) == NVME_CTRL_CTRATT_UUID_LIST) {
+			/* UUID list is supported.  Get the WDC UUID entry */
+			ret = nvme_identify_uuid(fd, &uuid_list);
+
+			if (ret == 0) {
+            	/* The 0th entry is reserved so start at 1 */
+				for (*uuid_ix = 1; *uuid_ix < NVME_MAX_UUID_ENTRIES; (*uuid_ix)++) {
+
+					/* The list is terminated by a zero UUID value */
+					if (memcmp(uuid_list.entry[*uuid_ix].uuid, zero_uuid, sizeof(zero_uuid)) == 0) {
+						/* if we reached the end then we did not find the uuid */
+						*uuid_ix = 0;
+						break;
+					} else {
+
+						if (memcmp(uuid_list.entry[*uuid_ix].uuid, uuid.b, sizeof(uuid.b)) == 0) {
```

Comment:
what is uuid.b? In my implementation, the uuid_t is just a u8[16] type, it doesn't contain a member called 'b'.

Reply:
I used the uuid_t as it is defined in nvme.h:
typedef struct {
	uint8_t b[16];
} uuid_t;
 But that is because LIBUUID is not defined.  I assume you would be expecting that library to be included so it will use a definition in uuid.h.  Where do I get libuuid?

Reply:
I see. This type should be treated as opaque: we can't peek into its implementation for portability reasons since we prefer to let the definition come from elsewhere. Use &uuid instead of referencing the member by name.

FWIW, you can get the appropriate library using your package manager's (dnf, yum, apt, zypper, etc...) install with libuuid-devel or uuid-dev.

Reply:
Ok got it.  I got uuid-dev installed also.